spotpass/3ds-task-finder.js

```diff
@@ -0,0 +1,73 @@
+const https = require('node:https');
+const axios = require('axios');
+const fs = require('fs-extra');
+const { COUNTRIES, LANGUAGES } = require('./constants');
+const apps = require('./ctr-boss-apps.json');
+
+const NPFL_URL_BASE = 'https://npfl.c.app.nintendowifi.net/p01/filelist';
+const TASKS = ['FGONLYT', 'news', 'data', 'TASK00', '0000001']; // * For this example, I'm using some common task names that are present in many games, but there are other ways this can be used, such as for matching tasks across different regions for a single game.
+
+const httpsAgent = new https.Agent({
+	rejectUnauthorized: false,
+	cert: fs.readFileSync('./certs/wiiu-common.crt'),
+	key: fs.readFileSync('./certs/wiiu-common.key'),
+});
+
+async function main() {
+	for (const app of apps) {
+		for (const task of TASKS) {
+			// * Skip the task if the app already has it
+			let titleHasTask = false;
+			for (const appTask of app.tasks) {
+				if (task.toLowerCase() == appTask.toLowerCase()) {
+					titleHasTask = true;
+					break;
+				}
+			}
+
+			if (titleHasTask) {
+				continue;
+			}
```

Comment:
Not really an issue, just a small suggestion for a possible refactor. You could remove the need for this `titleHasTask` variable, the check with it, and the `continue` statement by using a label on the `TASKS` loop and breaking to it in side the innermost loop.

It's fine as-is, just letting you know the option exists.

Reply:
Eh, it works, and ultimately this script isn't gonna be used beyond 2-3 weeks from now anyway so I don't think it matters much.